Sprint 05

YanikMax opened the pull request:
Логика работы приложения разбита на несколько частей.
Написан класс для ведения статистики, учитывается лучшая игра.
Алерт теперь предоставляет больше информации. Добавлены средняя точность и лучшая игра за все время приложения

YanikMax committed:
Задача 4 спринта. Добавил реализацию логики согласно макету в Figma

YanikMax committed:
Добавил блокировку кнопок Да и Нет при переходе на следующий вопрос.
Добавил скругление зеленой и красной рамок при ответе

YanikMax committed:
Добавил комментарий к коду о скруглении рамок

YanikMax committed:
Код с настройками скругления углов перенесен в ViewDidLoad
Лишний код с константой alert удален

YanikMax committed:
Логика работы приложения разбита на несколько частей
Реализован обновленный алерт с данными по игре
Ведется статистика по играм и сохраняется результат лучшей игры

Helvetii951 commented on MovieQuiz/Services/StatisticService.swift:
```diff
@@ -0,0 +1,78 @@
+    var gamesCount: Int { get }
+    var bestGame: GameRecord { get set } //здесь указал дополнительно set для сохранения значения
+    
+    func store(correct count: Int, total amount: Int)
```
Такие коменты лучше удалять. Так код будет легче читать, он будет чище.

Helvetii951 commented on MovieQuiz/Services/StatisticService.swift:
```diff
@@ -0,0 +1,78 @@
+        set {
+            self._gamesCount = newValue
+        }
+    }
```
Эти данные нужно будет хранить в `UserDefaults`, чтобы данные сохранялись между запусками приложения.

Helvetii951 commented on MovieQuiz/Presentation/AlertPresenter.swift:
```diff
@@ -0,0 +1,27 @@
+import UIKit
+
+class ResultAlertPresenter: AlertProtocol {
+    private let viewController: UIViewController
+    private let restartAction: (() -> Void)?
```
Здесь нужно будет сделать слабую ссылку.

YanikMax committed:
Исправлен код считающий среднюю точность ответов в квизе.
Добавлена слабая ссылка в классе ResultAlertPresenter переменной var viewController: UIViewController?
В userDefaults теперь сохраняются данные по gamesCount и totalAccuracy.

!!! Для Рената
Я не понимаю почему следует удалять комментарии. Да, код чище, правда не понимаю как это влияет на чтение, если он выделен серым. Вообще я бы с удовольствием добавлял комменты для себя в еще бОльшем количестве, потому что это помогает вспомнить всё, когда ты возвращаешься к коду через какое-то время.
Совершенно не понял про многострочный вариант строки, чтобы код умещался в столбец 120-180. Что есть столбцы? Это есть строки? Но у меня нет столько строк в данном коде.

Helvetii951 commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,72 +1,135 @@
+            let gamesCount = statisticService.gamesCount //количество сыгранных квизов
+            let formattedAccuracy = (String(format: "%.2f", statisticService.totalAccuracy)) //средняя точность
+
+            let message = "Ваш результат: \(correctAnswers)/10\nКоличество сыгранных квизов: \(gamesCount)\nРекорд: \(bestGame.correct)/10 (\(date))\nСредняя точность: \(formattedAccuracy)%"
+
```
Извиняюсь, в прошлый раз я оставил коммент про многострочный вариант строки не в том месте. Сделать можно примерно  так:
```
let text =
"""
Ваш результат: \(correctAnswers)/\(questionsAmount)
Количество сыгранных квизов: \(statisticService.gamesCount)
Рекорд: \(bestGameStats) (\(localizedTime))
Средняя точность: \(totalAccuracyPercentage)
"""
```
